src/log-entry-dashboard/log-entry-form/log-entry-form.component.tsx

```diff
@@ -20,124 +20,305 @@ import {
 import { withAuthenticationContext } from '../../_shared/contexts/authentication/with-authentication-context.component';
 import { LogEntry } from '../../_shared/interfaces/log-entry.interface';
 import { createLogEntry } from '../../_shared/services/log-entry.service';
-import { inputClass } from './log-entry-form.styles';
-import { panelClass } from '../log-entry-dashboard.styles';
+import { timeForm, buttonContainer } from './log-entry-form.styles';
+import { RadioChangeEvent } from 'antd/lib/radio';
+import { any } from 'prop-types';
 
 const FormItem = Form.Item;
+const RadioButton = Radio.Button;
+const RadioGroup = Radio.Group;
+const { TextArea } = Input;
+const SelectOption = Select.Option;
 
 interface LogEntryFormProps extends FormComponentProps {
   authContext: IAuthenticationContext;
 }
 
 class LogEntryFormComponent extends React.Component<LogEntryFormProps> {
+  state = {
+    categoryId: 0,
+    pointsCompeleted: 'notCompleted',
+  };
+
+  handleHourChange = (e?: number) => {
+    const {
+      form: { setFieldsValue },
+    } = this.props;
+
+    setFieldsValue({ logValue: 0 });
+  };
+
+  handleCategoryChange = (e: string) => {
+    this.setState({ categoryId: e });
+  };
+
+  handleRaidoChange = (e: RadioChangeEvent) => {
+    if (e.target.value === 'completed') {
+      this.setState({ pointsCompeleted: 'completed' });
+    } else if (e.target.value === 'notCompleted') {
+      this.setState({ pointsCompeleted: 'notCompleted' });
+    }
+  };
+
+  hasErrors = (fieldsError: any) => {
+    return Object.keys(fieldsError).some(field => fieldsError[field]);
+  };
+
   handleOnSubmit = (e: React.SyntheticEvent) => {
     e.preventDefault();
     const {
       authContext: { currentUser },
       form,
     } = this.props;
+
     form.validateFields((err, values) => {
       if (err) {
         return;
       }
 
       createLogEntry({
-        title: values['description'],
-        arePointsCompleted: values['completedPoints'],
+        title: values['title'],
+        arePointsCompleted: values['completedPoints'] || null,
         notes: values['notes'],
         date: values['logDate'],
         value: values['logValue'],
         categoryId: values['category'], //Categories.BillableTime,
-        subCategoryId: values['subcategory'], //SubCategories.Admin,
+        subCategoryId: values['subcategory'] || null, //SubCategories.Admin,
         updatedBy: currentUser ? currentUser.id : '',
         createdAt: new Date(),
         updatedAt: new Date(),
       });
+
+      form.resetFields();
     });
   };
 
-  render() {
+  handleClearForm = () => {
+    const { form } = this.props;
+    form.resetFields();
+  };
+
+  renderCategoriesSelectOptions = () => {
+    let keys = Object.keys(Categories).filter(
+      (key: any) => !isNaN(Number(Categories[key]))
```

Comment:
restrain from using `any` as much as possible because we lose type safety anytime we use it.  most times you can figure out a typing for what you need.  feel free to ask if you can figure one out.  in this case this should be `string`

Reply:
also, why are you doing this isNaN check?  were you getting some errors that lead you to do this?  seems there might be an upstream issue that needs addressed.